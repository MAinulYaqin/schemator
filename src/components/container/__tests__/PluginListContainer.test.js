import React from "react";
import { shallow } from "enzyme";
import configureStore from "redux-mock-store";

import ConnectedComponent, {
  PluginListContainer
} from "../PluginListContainer";

jest.mock("../../presentational/PluginList", () => "PluginList");

describe("PluginListContainer", () => {
  const setup = propOverrides => {
    const props = {
      items: [
        {
          id: "1",
          name: "Laravel",
          author: "Gattigaga Hayyuta Dewa",
          description: "Export to Laravel",
          license: "MIT",
          version: "0.1.0",
          icon: "icon.png"
        },
        {
          id: "2",
          name: "SQL",
          author: "Gattigaga Hayyuta Dewa",
          description: "Export to SQL",
          license: "GPL",
          version: "1.0.0",
          icon: "icon.png"
        }
      ],
      keyword: "Laravel",
      active: "2",
      onClickItem: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<PluginListContainer {...props} />);

    return {
      wrapper,
      props
    };
  };

  const setupConnected = () => {
    const mockStore = configureStore();
    const store = mockStore({
      plugins: [
        {
          id: "1",
          name: "Laravel",
          author: "Gattigaga Hayyuta Dewa",
          description: "Export to Laravel",
          license: "MIT",
          version: "0.1.0",
          icon: "icon.png"
        },
        {
          id: "2",
          name: "SQL",
          author: "Gattigaga Hayyuta Dewa",
          description: "Export to SQL",
          license: "GPL",
          version: "1.0.0",
          icon: "icon.png"
        }
      ]
    });

    store.dispatch = jest.fn();

    const wrapper = shallow(<ConnectedComponent store={store} />);

    return {
      wrapper,
      store
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should maps state to props", () => {
    const { wrapper, store } = setupConnected();
    const { plugins } = store.getState();

    expect(wrapper.props()).toMatchObject({ items: plugins });
  });
});
