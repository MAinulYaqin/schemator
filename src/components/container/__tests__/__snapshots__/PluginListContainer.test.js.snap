// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginListContainer should renders default 1`] = `
<PluginList
  active="2"
  items={
    Array [
      Object {
        "author": "Gattigaga Hayyuta Dewa",
        "description": "Export to Laravel",
        "icon": "icon.png",
        "id": "1",
        "license": "MIT",
        "name": "Laravel",
        "version": "0.1.0",
      },
      Object {
        "author": "Gattigaga Hayyuta Dewa",
        "description": "Export to SQL",
        "icon": "icon.png",
        "id": "2",
        "license": "GPL",
        "name": "SQL",
        "version": "1.0.0",
      },
    ]
  }
  keyword="Laravel"
  onClickItem={[Function]}
/>
`;
